### PR TITLE
chore(TDP-5900 -> TDP-5903): Pipeline performance improvements

### PR DESCRIPTION
**Pull request by fhuaulme**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5900
https://jira.talendforge.org/browse/TDP-5901
https://jira.talendforge.org/browse/TDP-5902
https://jira.talendforge.org/browse/TDP-5903

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [x] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/node/ReactiveTypeDetectionNode.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    public void receive(DataSetRow row, RowMetadata metadata) {
+        this.workingMetadata = metadata;
+        sink.emit(row.clone());
+    }
+
+    @Override
+    public void signal(Signal signal) {
+        try {
+            switch (signal) {
+            case END_OF_STREAM:
```
Comment (fabballe) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/node/ReactiveTypeDetectionNode.java]: why a switch with all the value ?

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/Flattener.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.talend.dataprep.transformation.pipeline;
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/Flattener.java]: 👍

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/Pipeline.java [hunk 3]
```diff
@@ -384,7 +378,13 @@ public Pipeline build() {
             current.to(outputSupplier.get());
             current.to(monitorSupplier.get());
 
-            pipeline.setNode(current.build());
+            final Node sourceNode = current.build();
+            pipeline.setNode(sourceNode);
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/Pipeline.java]: The optimization step is standalone and fully optional. Pipeline before optimization and after produce the same result.

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/actions/common/ActionsUtils.java [hunk 4]
```diff
@@ -91,8 +91,14 @@ public static void createNewColumn(ActionContext context, List<AdditionalColumn>
...
                 brandNewColumnBuilder.copy(newColumn).computedId(StringUtils.EMPTY);
                 brandNewColumnBuilder.type(Type.get(newColumn.getType()));
+                if (additionalColumn.typeSpecified()) {
+                    brandNewColumnBuilder.typeForce(true);
+                }
             } else {
                 brandNewColumnBuilder.type(additionalColumn.getType());
+                if (additionalColumn.typeSpecified()) {
+                    brandNewColumnBuilder.typeForce(true);
+                }
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/actions/common/ActionsUtils.java]: I wonder what you mean by this comment, will take this offline.

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/node/ReactiveTypeDetectionNode.java (hunk 1 above)
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/pipeline/node/ReactiveTypeDetectionNode.java]: 👍

**Pull request merged**